apicast/src/policy_chain.lua

```diff
@@ -0,0 +1,89 @@
+local setmetatable = setmetatable
+local insert = table.insert
+local error = error
+local rawset = rawset
+local type = type
+local require = require
+local noop = function() end
+
+local linked_list = require('linked_list')
+local policy = require('policy')
+
+local _M = {
```

Comment:
I think that this class is exposing lots of methods that it shouldn't expose. I think that only `build`, `export` and all the phases (access, rewrite, etc.) should be public. What do you think?

Reply:
Yep, good idea.

Reply:
Also, it'd be good to document the public methods. Specially, I think that the purpose of `export` is not obvious looking at this module in isolation.